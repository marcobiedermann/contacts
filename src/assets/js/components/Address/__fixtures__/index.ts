import { AddressProps } from '..';

const props: AddressProps = {
  city: 'New York City',
  country: 'USA',
  streetAddress: 'Main Street',
  zipCode: '10001',
};

export default props;
