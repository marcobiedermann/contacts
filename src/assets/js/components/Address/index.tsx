import React, { FC } from 'react';

export interface AddressProps {
  city: string;
  country: string;
  streetAddress: string;
  zipCode: string;
}

const Address: FC<AddressProps> = (props) => {
  const { city, country, streetAddress, zipCode } = props;

  return (
    <address>
      <a
        href={`https://maps.google.com/?q=${encodeURIComponent(streetAddress)} ${zipCode} ${city} ${country}`}
        target="_blank"
        rel="noopener noreferrer"
      >
        {streetAddress}
        <br />
        {zipCode} {city}
        <br />
        {country}
      </a>
    </address>
  );
};

export default Address;
