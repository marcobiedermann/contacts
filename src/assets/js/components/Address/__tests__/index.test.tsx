import { render } from '@testing-library/react';
import React from 'react';
import Address from '..';
import props from '../__fixtures__';

describe('address component', () => {
  it('should render correctly', () => {
    expect.assertions(1);

    const { asFragment } = render(<Address {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
