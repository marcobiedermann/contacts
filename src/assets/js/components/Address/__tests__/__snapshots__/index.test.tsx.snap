// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`address component should render correctly 1`] = `
<DocumentFragment>
  <address>
    <a
      href="https://maps.google.com/?q=Main%20Street 10001 New York City USA"
      rel="noopener noreferrer"
      target="_blank"
    >
      Main Street
      <br />
      10001 New York City
      <br />
      USA
    </a>
  </address>
</DocumentFragment>
`;
